crates/sui-indexer-restorer/src/archives.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::num::NonZeroUsize;
+
+use prometheus::Registry;
+use sui_types::digests::CheckpointDigest;
+use tracing::info;
+
+use sui_archival::reader::{ArchiveReader, ArchiveReaderMetrics};
+use sui_config::node::ArchiveReaderConfig;
+use sui_config::object_storage_config::{ObjectStoreConfig, ObjectStoreType};
+
+use crate::Args;
+
+#[derive(Clone, Debug)]
+pub(crate) struct ArchivalCheckpointInfo {
+    pub next_checkpoint_after_epoch: u64,
+    #[allow(unused)]
+    pub chain_identifier: CheckpointDigest,
+}
+
+pub(crate) async fn read_archival_checkpoint_info(
```

Comment:
- Can you add a doc comment here explaining what this is fetching?
- Also can you add some comments explaining the steps that this code is taking? e.g. why does the object store need to be configured this way, what does it mean to sync the manifest, why do we need a manifest, etc.
- I'm not seeing a place where we're using the chain identifier at the moment -- is that still necessary? (will it be used as part of some later PR?). In `sui-indexer-alt`, we have a bootstrapping process which means we can fetch the genesis information from the remote store URL, so we may not need to read data about the first checkpoint, unless the chain identifier is required for some other part of the bootstrapping process.
- The `ObjectStoreConfig` here is subtly different here, vs in the main restorer (one uses GCS, the other uses S3) this means that we're depending on both of these stores to be around for restoration to work, which seems kind of unfortunate. What's the background on that -- is it possible for us to use only one object store?
- The use of the `args.concurrency` parameter is a little strange here. IIUC, this code path fetches at most one thing, it might as well have a concurrency of 1? In general it seems like this function only needs to accept the `archive_bucket` and `start_epoch` from the args, and accepting those explicitly may make what is happening here clearer.
- Should this function be in an `impl` for `ArchivalCheckpointInfo`?
- nit: `anyhow::Result<ArchivalCheckpointInfo>` (it's just shorter).

Reply:
> The ObjectStoreConfig here is subtly different here, vs in the main restorer (one uses GCS, the other uses S3) this means that we're depending on both of these stores to be around for restoration to work, which seems kind of unfortunate. What's the background on that -- is it possible for us to use only one object store?

I am not happy with this either, the current mixed data sources are the only reliably usable ones I know of, would be good if I can fetch them from one source do not want to wait on that.

Reply:
> I'm not seeing a place where we're using the chain identifier at the moment

it was used for the chain identifier table and I just checked and will use this for kv_genesis in the new schema.

Reply:
The restorer doesn't need to worry about the `kv_genesis` table -- it is designed to be bootstrapped in one step by the indexer, regardless of which checkpoints indexing needs to resume from, so I think we can simplify this part.